geode-core/src/test/java/org/apache/geode/cache/query/internal/MethodDispatchTest.java

```diff
@@ -62,20 +70,180 @@ public void invokeShouldThrowExceptionWhenMethodIsNotAuthorized() throws NameRes
     MethodDispatch methodDispatch = new MethodDispatch(methodInvocationAuthorizer, TestBean.class,
         "publicMethod", Collections.emptyList());
 
-    assertThatThrownBy(() -> methodDispatch.invoke(testBean, emptyList))
+    assertThatThrownBy(() -> methodDispatch.invoke(testBean, emptyList, queryExecutionContext))
         .isInstanceOf(NotAuthorizedException.class)
         .hasMessage(RestrictedMethodAuthorizer.UNAUTHORIZED_STRING + "publicMethod");
   }
 
+  @Test
+  public void invokeShouldCacheResultForAllowedMethod() throws Exception {
+    doReturn(true).when(methodInvocationAuthorizer).authorize(any(), any());
+    MethodDispatch methodDispatch =
+        new MethodDispatch(methodInvocationAuthorizer, Integer.class, "toString",
+            Collections.emptyList());
+    methodDispatch.invoke(new Integer("0"), emptyList, queryExecutionContext);
+
+    assertThat(queryExecutionContext.cacheGet("java.lang.Integer.toString")).isNotNull();
```

Comment:
Do we need to assert both that it's not null and that it's true?

Reply:
👍